### PR TITLE
Vcov vignette

### PR DESCRIPTION
**Pull request by nathanielolin**
Adding use case for variance-covariance matrix to usage vignette

### CODE REVIEW

File: R/mod_marg.R [hunk 1]
```diff
@@ -1,33 +1,55 @@
...
+#' since \code{poly} needs more data to calculate orthogonal polynomials
+#' (e.g. \code{poly(10, 2)} fails, but \code{poly(c(10, 8, 3), 2)} will run).
+#'
+#' P values are calculated with T tests for gaussian families, and Z tests
+#' otherwise. If a new variance-covariance matrix is provided (e.g. for
+#' clustering standard errors), the degrees of freedom for the T test / p-value
+#' calculation may need to be specified using dof. To replicate Stata clustering
+#' \code{vce(cluster var_name)}, dof should be set to \eqn{g - 1}, where g is
+#' the number of unique levels of the clustering variable.
 #'
```
Comment (nathanielolin) [inline R/mod_marg.R]: I think some documentation changes are getting pulled in from master here.

File: R/mod_marg.R [hunk 2]
```diff
@@ -73,6 +95,11 @@ mod_marg2 <- function(mod, var_interest,
   data <- data[, names(data) %in% all.vars(mod$formula)]
   data <- data[complete.cases(data), ]
 
+  if(sum(grepl("poly\\(.*\\)", names(mod$model))) !=
+     sum(grepl("raw = T", names(mod$model))))
+    warning(paste("If you're using 'poly()' for higher-order terms,",
+                  "use the raw = T option (see ?poly)"))
+
```
Comment (nathanielolin) [inline R/mod_marg.R]: Ditto

File: vignettes/usage.Rmd [hunk 3]
```diff
@@ -101,7 +101,42 @@ g <- glm(y ~ poly(distance, 2, raw = T)*as.factor(agegroup) , data = margex)
...
+(conditional on covariates) is constant across our sample. Sometimes, this assumption 
+is violated, and we will use a different variance-covariance matrix to represent this 
+heterogeneity in variance (heteroskedasticity). Creating these variance-covariance 
+matrices is beyond the scope of this package. However, they can be used with `mod_marg2`
+to correct standard errors and p values in predicted levels or effects.
+
+Let's say we want to get the predicted levels of an outcome for different treatments,
+but we want to cluster our standard errors by the `arm` variable. We estimate the model,
+and then create the "clustered" variance-covariance matrix separately 
+(see `modmarg/data-raw/make_cluster_vcov.R` for one method to do this).
```
Comment (anniejw6) [inline vignettes/usage.Rmd]: I'm not sure users will be able to find this. I would consider linking to github.

Also I'd be explicit that this replicates Stata's methods for clustering.

File: vignettes/usage.Rmd [hunk 4]
```diff
@@ -101,7 +101,42 @@ g <- glm(y ~ poly(distance, 2, raw = T)*as.factor(agegroup) , data = margex)
 summary(g)
 
 unique(margex$agegroup)
-mod_marg2(mod = g, var_interest = "agegroup", type = 'levels' 
+mod_marg2(mod = g, var_interest = "agegroup", type = 'levels',
           at_var_interest = c("20-29"))
 ```
 
+# Use Case 4: Getting Margins with Different Variance-Covariance Matrices
+Normally we assume that the amount of variation in our outcome of interest
```
Comment (anniejw6) [inline vignettes/usage.Rmd]: "Normally" : is that a pun?

File: R/mod_marg.R (hunk 1 above)
Comment (anniejw6) [inline R/mod_marg.R]: fwiw, best practice if you're forking off a branch is to check out that branch (Branch A), check out another branch (Branch B) from that branch, and then make a PR from Branch B back to Branch A.

**Comment by nathanielolin** [comment]
I did do that, but when I checked out Branch A, my local branch had all the new commits from Master which hadn't / haven't been merged into Branch A yet.

File: vignettes/usage.Rmd (hunk 4 above)
Comment (nathanielolin) [inline vignettes/usage.Rmd]: :-| not intentionally.

**Comment by anniejw6** [comment]
@nathanielolin - it's a little dangerous to be editing files other than the one you intend to. I'm going to close this PR. Then, can you make a PR that just updates vignette?

**Pull request closed without merging**